src/main/java/org/mule/extension/http/api/HttpRequestAttributesBuilder.java

```diff
@@ -139,7 +142,36 @@ public HttpRequestAttributes build() {
     requireNonNull(requestUri, "Request URI cannot be null.");
     requireNonNull(localAddress, "Local address cannot be null.");
     requireNonNull(remoteAddress, "Remote address cannot be null.");
-    return new HttpRequestAttributes(headers, listenerPath, relativePath, version, scheme, method, requestPath, requestUri,
+    this.proxyRequestPath = resolveProxyRequestPath();
```

Comment:
I think you are recalculating the proxyRequestPath for each `build` but it may already be present. We should find a way to avoid this. Perhaps calculating it whenever the listenerPath or requestPath change and the other is present?